tests/test_tictactoe.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+import numpy as np
+import random
+from metagamer.environments import tictactoe
+
+
+@pytest.fixture
+def tt():
+    return tictactoe.TicTacToeEnv()
+
+
+def test_simple_game(tt):
+    _, reward, done, _ = tt.step((0, 0))
+    assert not done
+    assert reward == 0
+    _, reward, done, _ = tt.step((0, 1))
+    assert not done
+    assert reward == 0
+    _, reward, done, _ = tt.step((1, 0))
+    assert not done
+    assert reward == 0
+    _, reward, done, _ = tt.step((1, 1))
+    assert not done
+    assert reward == 0
+
+    _, reward, done, _ = tt.step((2, 0))
+    assert done
+    assert reward == 1
+
+
+@pytest.mark.parametrize("player", (1, -1))
+def test_diagonal(player):
+    board = np.zeros((3, 3))
+    board[0, 0] = player
+    assert tictactoe.check_win(board) == 0
+    board[1, 1] = player
+    assert tictactoe.check_win(board) == 0
+    board[2, 2] = player
+    assert tictactoe.check_win(board) == player
+
+
+@pytest.mark.parametrize("player", (1, -1))
+def test_rev_diagonal(player):
+    board = np.zeros((3, 3))
+    board[0, 2] = player
+    assert tictactoe.check_win(board) == 0
+    board[1, 1] = player
+    assert tictactoe.check_win(board) == 0
+    board[2, 0] = player
+    assert tictactoe.check_win(board) == player
+
+
+@pytest.mark.parametrize("seed", range(100))
+def test_random(tt, seed):
+    random.seed(seed)
+    while not tt.done:
+        action = random.choice(tt.valid_actions)
+
+        _, reward, *_ = tt.step(action)
```

Comment:
I'm not sure about these underscores and stars here in this particular line.

Reply:
so you can do something like
`a,b,c,d = range(4)`
and a =0, b = 1, c = 2, d =3 

If you only wanted the second value, you don't need to keep giving variables, you can use the * to indicate you want more than a single var (you'll see this in *args)
`a,b,*c = range(4)`
a = 0, b=1, c= (2,3)

Now it's convention to use underscore for things you don't care about. If I saved all the names, you might feel the need to commit them to memory and assume they're used a bit later. underscores basically allow you to ignore the rest instantly.

`_, b, *_ = range(4)`
b = 1

Reply:
Ah, so _ os almost like a blank throwaway thing for things you need to have a space for when unpacking, but don't intend to use after you've unpacked something.  Thanks for the explanation!

Reply:
I'm not adding any doc here since this a python specific standard